server.js

```diff
@@ -0,0 +1,118 @@
+const fs = require('fs');
+
+const bodyParser = require('body-parser');
+const cv = require('opencv4nodejs');
+const express = require('express');
+const Ziggeo = require('ziggeo');
+const ZiggeoSdk = new Ziggeo(process.env.ZIGGEO_APP_TOKEN, process.env.ZIGGEO_PRIVATE_KEY, process.env.ZIGGEO_ENCRYPTION_KEY);
+
+const smileClassifier = new cv.CascadeClassifier(cv.HAAR_SMILE);
+const faceClassifier = new cv.CascadeClassifier(cv.HAAR_FRONTALFACE_DEFAULT);
+const port = process.env.PORT || 3000;
+const smileThreshold = 3.6;
+
+const app = require('express')();
+app.use(bodyParser.json()); // for parsing application/json
+
+// post route for processing videos.
+// expected body: {'videoToken': as7sd79d}.
+// result json: {'videoToken': as7sd79d}.
+app.post('/process-video', function (req, res) {
+    token = req.body.videoToken;
+    annotatedVideo = '';
+    ZiggeoSdk.Videos.download_video(token, {
+        success: function(result) {
+            fs.writeFile(token + '.mp4', result, {}, (err, result) => {
+                if(err){
+                    res.json(500, 'failed to write video');
+                    return;
+                }
+                annotatedVideo = annotateSmile(token + '.mp4');
+                ZiggeoSdk.Videos.create({
+                    file: annotatedVideo,
+                }, {
+                    success: function(result) {
+                        fs.unlinkSync(token + '.mp4');
+                        fs.unlinkSync(annotatedVideo);
+                        res.json({videoToken: result.token});
```

Comment:
so if smile/annotated video exists, create a video. otherwise, send me back the old video token so that i can use it on the client side to test if our SmileBot is going to say "yes" or "no" to there being a smile in the video.

Reply:
excellent!